src/ContinuumAbsorption/absorption_metals_bf.jl

```diff
@@ -36,8 +36,11 @@ function metal_bf_absorption!(α, νs, T, number_densities)
             end
             # cross-section interpolator
             σ_itp = metal_bf_cross_sections[spec]
-            α .+= exp.(log(number_densities[spec]) .+ σ_itp.(νs, log10(T))) * 1e-18 #convert to cm^2 
+            σ = σ_itp.(νs, log10(T))
+            # Protecting this is necessary to avoid NaNs in derivatives.  
+            # It's probably better to avoid them in the first place for performance reasons.
+            mask = isfinite.(σ)
```

Comment:
what are the cases where σ is not finite? Outside of the interpolation table?

Reply:
Those are -Infs, which translate to 0s after being exp'd.  That process turned the derivative part of the dual numbers into NaNs.

Reply:
I realize that there are likely better ways to do this, but this produces correct results, and I don't think it's worth redoing unless this proves to be a speed bottleneck.

Reply:
That perfectly sensible to me!

It might be worth adding a line to the comments explaining where the NaNs come from (since the source non-obvious and presumably took you quite a bit of work to figure out)

Reply:
done.